### PR TITLE
ARROW-10905: [Python] Add support for Python 3.9 Windows wheels

### PR DESCRIPTION
**Pull request by terencehonles**
Picking back up from #8386 in case I can handle this or someone else wants to work off of it. Likely will need to be updated based on #8881 / #8778.

### CODE REVIEW

**Comment by nealrichardson** [comment]
#8881 has merged now so I think this needs to be rebased.

**Comment by terencehonles** [comment]
> #8881 has merged now so I think this needs to be rebased.

Thanks for the heads up @nealrichardson

**Comment by terencehonles** [comment]
backing out Sutou Kouhei's changes temporarily to see where things stand without the subsequent modifications.

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp38 wheel-win-cp39 conda-win-vs2017-py39

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp39

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp39

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp39

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp39

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m wheel-win-cp37m wheel-win-cp38

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-* conda-win-*

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m

File: dev/tasks/python-wheels/win-build.bat [hunk 1]
```diff
@@ -22,11 +22,19 @@
 
 @rem --file=arrow\ci\conda_env_gandiva.yml ^
 
-@rem create conda environment for compiling
 @rem FIXME: Update to numpy 1.16.6
+SET NUMPY_VERSION=1.16.5
+SET VS_VERSION=
+if %PYTHON_VERSION% == 3.6 (
+    SET VS_VERSION="<14.16"
```
Comment (kou) [inline dev/tasks/python-wheels/win-build.bat:29]:
```suggestion
    SET VS_VERSION=<14.16
```

Comment (terencehonles) [reply dev/tasks/python-wheels/win-build.bat:29]: That's what I had here https://github.com/ursa-labs/crossbow/runs/1672510076 and it looked like it was treating it as interpolation. I'm not sure if it's not liking the multiline or the `else if`.

Comment (terencehonles) [reply dev/tasks/python-wheels/win-build.bat]: I'm glad I don't need to deal with batch scripting on a regular basis, but it seems like this should have been `SET VS_VERSION=^<14.16` based on https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/set_1#remarks

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-* conda-win-*

**Comment by terencehonles** [comment]
@xhochy when you get a chance it looks like #8881 did not fix building on Python 3.6 as [with my change](https://github.com/ursa-labs/crossbow/runs/1672835933) and [without my change (master)](https://github.com/ursa-labs/crossbow/runs/1674578035) both seem to fail on finding OpenSSL.

I'll try to see if I can poke around further, but not sure how far I can get.

**Comment by terencehonles** [comment]
@kou any chance you know what to do about:

> ```
>   File "arrow/dev/tasks/crossbow.py", line 1662, in upload_artifacts
>     tag_name=tag, target_commitish=sha, method=method, patterns=patterns
>   File "arrow/dev/tasks/crossbow.py", line 460, in github_overwrite_release_assets
>     repo = self.as_github_repo()
>   File "arrow/dev/tasks/crossbow.py", line 384, in as_github_repo
>     self._github_repo = github.repository(username, reponame)
>   File "C:\Miniconda\lib\site-packages\github3\github.py", line 1981, in repository
>     json = self._json(self._get(url), 200)
>   File "C:\Miniconda\lib\site-packages\github3\models.py", line 156, in _json
>     raise exceptions.error_for(response)
> github3.exceptions.AuthenticationFailed: 401 Bad credentials
> ```

from https://dev.azure.com/ursa-labs/crossbow/_build/results?buildId=22650&view=logs&j=4c86bc1b-1091-5192-4404-c74dfaad23e7&t=ec99a26b-0264-5e86-36fb-9cfd0ca0f9f3&l=37 

I probably didn't need to rebuild everything since that step will probably fail across the board (I'll check each of them since I hadn't done that in https://github.com/apache/arrow/pull/8916#issuecomment-757096365 )

**Comment by xhochy** [comment]
> @xhochy when you get a chance it looks like #8881 did not fix building on Python 3.6 as [with my change](https://github.com/ursa-labs/crossbow/runs/1672835933#step:6:4975) and [without my change (master)](https://github.com/ursa-labs/crossbow/runs/1674578035#step:6:4908) both seem to fail on finding OpenSSL.
> 
> I'll try to see if I can poke around further, but not sure how far I can get.

The OpenSSL issue is not problem. This just skips the SSL-supporting part of thrift, we don't need that. I actually fail to parse the log to find what is actually the issue though.

**Comment by xhochy** [comment]
> > @xhochy when you get a chance it looks like #8881 did not fix building on Python 3.6 as [with my change](https://github.com/ursa-labs/crossbow/runs/1672835933#step:6:4975) and [without my change (master)](https://github.com/ursa-labs/crossbow/runs/1674578035#step:6:4908) both seem to fail on finding OpenSSL.
> > I'll try to see if I can poke around further, but not sure how far I can get.
> 
> The OpenSSL issue is not problem. This just skips the SSL-supporting part of thrift, we don't need that. I actually fail to parse the log to find what is actually the issue though.

OpenSSL is an issue but for gRPC!

File: dev/tasks/python-wheels/win-build.bat [hunk 2]
```diff
@@ -22,11 +22,21 @@
...
+) else if %PYTHON_VERSION% == 3.9 (
+    SET NUMPY_VERSION=1.19.5
+)
+
+@rem create conda environment for compiling
 call conda.bat create -n wheel-build -q -y -c conda-forge ^
-    cmake ninja pkg-config numpy=1.16.5 ^
-    "vs2015_runtime<14.16" ^
+    cmake ninja pkg-config numpy=%NUMPY_VERSION% ^
```
Comment (xhochy) [inline dev/tasks/python-wheels/win-build.bat:38]: Can you try this?
```suggestion
    cmake ninja pkg-config numpy=%NUMPY_VERSION% libopenssl-static ^
```

**Comment by kou** [comment]
> @kou any chance you know what to do about:

It seems that we need to update a secret variable registered in https://dev.azure.com/ursa-labs/crossbow/ but I don't have permission to update it. It seems that we don't care about `conda-win-*` tasks in this pull request because we don't touch anything for `conda-win-*` in this pull request.

**Comment by kou** [comment]
It may be better that we wait for https://github.com/apache/arrow/pull/9096 because it uses vcpkg instead of conda.
It doesn't have the OpenSSL problem: https://github.com/apache/arrow/pull/9096#issuecomment-756821064

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m wheel-win-cp39

**Comment by terencehonles** [comment]
> It may be better that we wait for #9096 because it uses vcpkg instead of conda.
> It doesn't have the OpenSSL problem: [#9096 (comment)](https://github.com/apache/arrow/pull/9096#issuecomment-756821064)

Ok, I'll just test out the suggestion. Not sure when @kszucs plans on trying to get #9096 in.

**Comment by kszucs** [comment]
> > @kou any chance you know what to do about:
> 
> It seems that we need to update a secret variable registered in https://dev.azure.com/ursa-labs/crossbow/ but I don't have permission to update it. It seems that we don't care about `conda-win-*` tasks in this pull request because we don't touch anything for `conda-win-*` in this pull request.

I updated the secret and also invited you to the project.

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-win-cp36m

**Comment by kszucs** [comment]
Superseded by #9096.

**Pull request closed without merging**